### DISCUSSION TITLE
Is it possible to encrypt all the websocket data traffic shown in the browser?

### DISCUSSION THREAD

**Discussion by babashark** [nats-io/nats-server#5680] [Q&A]
I can see all the nats protocol message in the browser
![image](https://github.com/user-attachments/assets/5a96bd1d-317b-4a48-b423-d9ab54a4d89f)

but I want to do something like this
![image](https://github.com/user-attachments/assets/60174306-4101-4b29-85fe-2c34265ade30)

**Comment by derekcollison** [comment]
You can encrypt the transport for websockets of course but the browser will decrypt as part of the transport.

If you want only your app to see the contents and decrypt than you need to do that at the app layer.
